e2e/issues_test.go

```diff
@@ -0,0 +1,81 @@
+// +build e2e
+
+//
+
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/hashicorp/consul-terraform-sync/controller"
+	"github.com/hashicorp/consul-terraform-sync/event"
+	"github.com/hashicorp/consul-terraform-sync/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+// issues GH-236 and GH-239
+func TestServicesRenderRace(t *testing.T) {
+	const NumberOfServices = 100
+
+	srv := testutils.NewTestConsulServer(t,
+		testutils.TestConsulServerConfig{HTTPSRelPath: "../testutils"})
+	defer srv.Stop()
+
+	tempDir := t.Name()
+	cleanup := testutils.MakeTempDir(t, tempDir)
+	//_ = cleanup
+	defer cleanup()
+
+	ctx, ctxCancel := context.WithCancel(context.Background())
+	defer ctxCancel()
+
+	// seed server with test data
+	testServices := testutils.TestServices(NumberOfServices)
+	testutils.AddServices(t, srv, testServices)
+
+	// config
+	conf := config.DefaultConfig()
+	conf.BufferPeriod.Enabled = config.Bool(false)
+	serviceNames := make([]string, NumberOfServices)
+	for i, s := range testServices {
+		serviceNames[i] = s.Name
+	}
+	conf.Tasks = &config.TaskConfigs{
+		&config.TaskConfig{
+			Name:     config.String("serv_rend_race_task"),
+			Source:   config.String("../../test_modules/null_resource"),
+			Services: serviceNames,
+		}}
+	conf.Consul.Address = config.String(srv.HTTPAddr)
```

Comment:
I'm wondering why HTTP is used for this e2e test instead of HTTPS, can you share some thought around this?

Reply:
It wasn't relevant to the specific test and the test was a bit simpler and ran a bit faster without.
We have a test for the HTTPS setup and so, IMHO, should otherwise use whichever protocol is faster or more appropriate for that specific test.

Reply:
👍 thanks fine with me then! Seeing how the server setup above used `testutils.NewTestConsulServer()` which adds the overhead of HTTPS stuff, I wanted to double check on the intent. 

Non-blocking comment: In this case you could optionally configure the server using the Consul SDK directly or make a helper function that only does HTTP.

```
srv, err := testutil.NewTestServerConfigT(tb,
	func(c *testutil.TestServerConfig) {
		c.LogLevel = "warn"
		c.Stdout = ioutil.Discard
		c.Stderr = ioutil.Discard
	})
```
https://github.com/hashicorp/consul-terraform-sync/blob/master/testutils/consul.go#L32-L36

Reply:
Using `testutils.NewTestConsulServer()` and having it support HTTPS doesn't really add much overhead to the startup. HTTPS mainly adds overhead to the connections from the TLS negotiation.
I think the common function works fine even if we don't use the TLS connection.